### PR TITLE
PW-1273 Undefined index check in cc payment response PW-1349 getCurrentLocaleCode fallback bug

### PR DESCRIPTION
**Pull request by cyattilakiss**
**Description**
Fix for unhandled result in the query parameters
Fix for billing agreement undefined index in case of api response
misconfiguration
Fix undefined $store variable

### CODE REVIEW

File: Model/Billing/Agreement.php [hunk 1]
```diff
@@ -179,36 +179,41 @@ public function setCcBillingAgreement($contractDetail, $storeOneClick, $storeId)
             ->setMethodCode('adyen_oneclick')
             ->setReferenceId($contractDetail['recurring.recurringDetailReference']);
 
+        if (!isset($contractDetail['cardBin']) ||
+            !isset($contractDetail['cardHolderName']) ||
+            !isset($contractDetail['cardSummary']) ||
+            !isset($contractDetail['expiryDate']) ||
+            !isset($contractDetail['paymentMethod'])
+        ) {
+            $this->_errors[] = __('"In the Additional data in API response section, select: Card summary, Expiry Date, Cardholder name, Recurring details and Variant to create billing agreements immediately after the payment is authorized."');
```
Comment (rikterbeek) [inline Model/Billing/Agreement.php]: where is this being used ?

File: Controller/Process/Result.php [hunk 2]
```diff
@@ -214,13 +214,18 @@ protected function _validateUpdateOrder($order, $response)
     {
         $result = false;
 
-        $this->_adyenLogger->addAdyenResult('Updating the order');
-
```
Comment (rikterbeek) [inline Controller/Process/Result.php]: don't we want to keep this log so we now the order updating process is started ?

Comment (cyattilakiss) [reply Controller/Process/Result.php]: I put it after the check for the result query param so it won't make any confusion that we are updating the order while we have already jumped out of it.

https://github.com/Adyen/adyen-magento2/pull/453/files#diff-bd7cef8f0429f77c78243ef5e37f68a5R228

File: Model/Billing/Agreement.php (hunk 1 above)
Comment (cyattilakiss) [reply Model/Billing/Agreement.php]: We show it to the user here:
https://github.com/Adyen/adyen-magento2/pull/453/files#diff-735f70c1f948ebbd90915b1548d93432R1631

This is shown at the magento backend under the order

Comment (rikterbeek) [reply Model/Billing/Agreement.php]: thanks clear

**Pull request merged**